### ISSUE TITLE
Write extract of on-order bib records from Symphony

### ISSUE THREAD

**Issue by dlrueda** [sul-dlss/FOLIO-Project-Stanford#110]
Depends on #99

Once Vitus/Suzette have confirmed which “on-order” bibs need to be loaded to FOLIO, write re-runnable selection of these bibs from Symphony.

I think best thing to do is generate a file of bibs, put in /s/SUL/Dataload/folio_symphony_extract/Bibs_to_load
AND 
generate csv file of "items” (right, they don’t exist in Symphony, so this is a file of items to be created in FOLIO for these on-order bibs) with ON-ORDER current location, and they can be loaded through airflow process with correct library, status ON-ORDER.  Barcode can be blank.

Airflow code should be able to create bibs from the .mrc file.  Need to create a mapping that will be able to handle the .csv file to correctly build holdings and items in FOLIO

**Comment by shelleydoljack** [comment]
Darsi mentioned that we might want to get the ckeys from the orders that are identified for migration. The files `/s/SUL/Dataload/folio_symphony_extract/acquisitions/orders/tsv/orders_sul.tsv` and `/s/SUL/Dataload/folio_symphony_extract/acquisitions/orders/tsv/orders_law.tsv` have CKEY as first field. These are records identified as on order and to be migrated to FOLIO using the scripts in `/s/SUL/Bin/folio_symphony_extract/acquisitions/orders`. It looks like maybe we just need to run run_reports.ksh to generate new files of orders to be migrated. Then write something to get the CKEYS from the orders_sul.tsv and orders_law.tsv files to put in a ckey list for the script do_catdumps_and_items_ckeylist_tsv.ksh.

**Comment by dlrueda** [comment]
I realized this leaves out Business since we are not migrating any orders for them.  Need to ask Kent about this

**Comment by shelleydoljack** [comment]
After conversation with Darsi, the plan is to use the ckeys from the orders_sul/law.tsv files and get the marc for those. Then we will create item tsv using data from the orders_sul/law.tsv file for the migration script. Here is a rough mapping of item tsv and the data pulled from orders_sul/law.tsv:

| CATKEY |CALL_NUMBER_TYPE|BASE_CALL_NUMBER |VOLUME_INFO|BARCODE|LIBRARY|HOMELOCATION|CURRENTLOCATION|ITEM_TYPE
|---|---|---|---|---|---|---|---|---|
|CKEY|?|?|PARTS_IN_SET?|blank|derived from HOLDNG_CODE|derived from HOLDNG_CODE|ON-ORDER|derived from HOLDNG_CODE|

@ahafele what should we do about call number? we can get it using orderline key that's in our orders data. but i think it will need to be the first orderline (there could be multiple orderlines we are creating in FOLIO).

**Comment by ahafele** [comment]
@shelleydoljack we would like to leave the call number blank for these.

**Comment by shelleydoljack** [comment]
Script is `/s/SUL/Bin/folio_symphony_extract/acquisitions/on_order_bibs/run_on_order_extract`. It calls a ksh and rb script that writes files to `/s/SUL/Dataload/folio_symphony_extract/Bibs_catdumps`: ON-ORDER.sul.mrc, ON-ORDER.sul.tsv, ON-ORDER.law.mrc, and ON-ORDER.law.tsv.

We still need to figure out about Business library on order bib records.

**Comment by shelleydoljack** [comment]
Done. Created new ticket for the Business library on order bibs: https://github.com/sul-dlss/FOLIO-Project-Stanford/issues/159

**Comment by dlrueda** [comment]
Kent says his on-order records are all ordered from Gobi, and a bib record is loaded for each of these and item created with current location of ON-ORDER, and a “real” home location.  So these bibs will get migrated on the basis of their home location, no need to do anything special for Business on order records.
